### PR TITLE
Update to separate call API

### PR DESCRIPTION
**Pull request by levb**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,6 +23,11 @@ build:
 run: build
 	LOCAL=true ./dist/mattermost-app-test ${BASE} ${ADDR}
 
+.PHONY: run-with-invalid
+## run-with-invalid: runs the service, includes invalid data
+run-with-invalid: build
+	LOCAL=true INCLUDE_INVALID=true ./dist/mattermost-app-test ${BASE} ${ADDR}
```
Comment (hanzei) [inline Makefile:29]: 0/5 switching the mode could be a slash command instead

File: other.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+	return apps.Binding{
+		Label: "other",
+		Bindings: []apps.Binding{
+			b,
+		},
+	}
+}
+
+func initHTTPOther(r *mux.Router) {
```
Comment (hanzei) [inline other.go:28]: Is there a more meaningful name then `Other`? Maybe
```suggestion
func initHTTPDialog(r *mux.Router) {
```

File: test_invalid.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
+package main
```
Comment (hanzei) [inline test_invalid.go:1]: the `test` in the file path let's me think that this is a test file. Maybe stick with `invalid.go`

**Review by hanzei — APPROVED** [review]
I didn't do a full testing of the app and only checked the code. The new structure LGTM. A few comments on naming.

File: other.go (hunk 2 above)
Comment (levb) [reply other.go:28]: 2/5 just drop this interactive dialog stuff. @larkox what was the value of the Interactive Dialog tests here?

**Review by hanzei — APPROVED** [review]

**Comment by levb** [comment]
Merging now to simplify testing of https://github.com/mattermost/mattermost-plugin-apps/pull/296

**Pull request merged**